HW_10/task_1.ts

```diff
@@ -0,0 +1,70 @@
+function getFirstElement<T>(arr: T[]): T {
+    return arr[0];
+}
+
+console.log(getFirstElement<string>(['1', '2']))
+console.log(getFirstElement<number>([1, 2]))
+
+interface IPerson {
+    name: string;
+    surname: string;
+    experienceOfYears: number;
+    getDetails(): string;
+}
+
+abstract class Employee implements IPerson {
+    protected salary = 0;
+    constructor(
+        public name: string,
+        public surname: string,
+        public experienceOfYears: number
+    ) {};
+
+    abstract getDetails(): string;
+    protected abstract calculateSalary(): number;
+}
+
+type TMethodology = 'Scrum' | 'Kanban';
+
+class Manager extends Employee{
+    constructor(
+        name: string,
+        surname: string,
+        experienceOfYears: number,
+        public prefered: TMethodology

```

Comment:
recommend to use private modificator if you don't use the field outside the class